bindings/python/benches/test_tiktoken.py

```diff
@@ -82,24 +82,30 @@ def benchmark_batch(model: str, documents: list[str], num_threads: int) -> None:
 def test(model: str, dataset: str, dataset_config: str, threads: List[int]):
     dataset_xnli = load_dataset(dataset, dataset_config)
 
-    input_lengths = [(10, False), (10_000, False), (10_000, True)]  # Example input lengths
+    # input_lengths = [(10, False), (10_000, False), (10_000, True)]  # Example input lengths
+    input_lengths = [(10_000, False, True), (10_000, False, False)]
 
     for num_threads in threads:
-        for length, fuse in input_lengths:
+        for length, fuse, long in input_lengths:
             documents = []
             for i, item in enumerate(dataset_xnli["train"]):
                 if i >= length:
                     break
-                documents.append("".join(item["premise"].values()))
+                if long:
+                    documents.append("".join(item["premise"].values()))
```

Comment:
long is non english only, good that you added the average length as well

Reply:
Yes but the principal factor affecting speed seems to be document length (I tried user arabic and had similar results)